apps/parsley/src/hooks/useSections/index.ts

```diff
@@ -108,32 +116,37 @@ export const useSections = ({
     [],
   );
 
+  /**
+   * This function will update the current section state and returns the next section state where
+   * all sections that contain the lineNumber values(s) are open. If the next state is the same as the current state,
+   * the function will return true and false otherwise.
+   * @param param0 is an object with a lineNumber key(s).
+   * @param param0.lineNumber is a number or an array of numbers that represent raw log line numbers.
+   * @returns true if the sectionState was updated and false otherwise
+   */
+  const openSectionContainingLineNumber = ({
+    lineNumber,
+  }: {
+    lineNumber: number | number[];
+  }): boolean => {
+    if (!sectionData || !sectionState) {
+      return false;
+    }
+    const nextState = openSectionContainingLineNumberHelper({
```

Comment:
This logic could actually exist inside of the `openSectionContainingLineNumber` hook function but it was too big and I wanted to test it. It's not really meant to be used outside of that hook function so that's why I added `...Helper`. I think `calculateNextSectionState` wouldn't capture that this function will return the next state with an open section based on the input line number. 
There is a function called `populateSections` and I thought about combining these two but I think it may complicate use cases and testing.